server.py

```diff
@@ -79,17 +122,27 @@ def home():
 @app.get("/dashboard")
 @login_required
 def dashboard():
+    contests = Contest.select().order_by(Contest.start_time)
     return bottle.template("dashboard.html", contests=contests)
 
 
 @app.get("/contest/<code>/<number>")
 @login_required
-def contest(code, number):
-    if not code in contests:
-        return "Contest does not exist"
-    if contests[code].start_time > datetime.datetime.now():
+def question(code, number):
+    if (
+        not ContestProblems.select()
+        .where(
+            (ContestProblems.contest == code)
+            & (ContestProblems.question == int(number))
+        )
+        .exists()
+    ):
+        return error404(404)
+    contest = Contest.get(Contest.code == code)
+    if contest.start_time > datetime.datetime.now():
         return "The contest had not started yet."
-    statement = questions[number].statement
+    with open(os.path.join(question_dir, number, "statement.txt"), "rb") as fl:
+        statement = fl.read()
```

Comment:
contest.description?

Reply:
This is the individual question page, we do not need the contest description here. Contest description is needed for `contest/<code>`

Reply:
sorry. my mistake. I thought the statement was the contest description.